google/api_versions.go

```diff
@@ -25,3 +31,63 @@ func Convert(item, out interface{}) error {
 
 	return nil
 }
+
+type TerraformResourceData interface {
+	HasChange(string) bool
+	GetOk(string) (interface{}, bool)
+}
+
+// Compare the keys set in schema against a list of features and their versions to determine
+//what version of the API is required in order to manage the resource.
+func getComputeApiVersion(d TerraformResourceData, baseVersion ComputeApiVersion, keys []Key) ComputeApiVersion {
```

Comment:
A resource that isn't in `v1` will have a base version higher than `v1`; it's the version of the resource itself. Renamed to `resourceVersion` to better show that.

I could also remove it entirely; it will only really be useful in the case that we have a `v0beta` resource where we also support some `v0alpha` fields. Let me know your thoughts.